### PR TITLE
Add chained fallback support and performance improved

### PR DESCRIPTION
**Pull request by varphone**
# New:
- Add more than one fallback with priority support, eg: `i18n!("locales", fallback = ["en", "es]);` 
  - #40
# Improved:
- Remove `RwLock` from `locale()` and `set_locale()`.
- String patterns replacement, time reduce 10% ~ 80%.
- Reduce memory copy on `t!()`.
# Breaking Changes:
- `rust_i18n::locale() -> String` => `rust_i18n::locale() -> Arc<String>` .
- `t!() -> String` => `t!() -> Cow<str>`.

### CODE REVIEW

**Comment by sunli829** [comment]
```
rust_i18n::locale() -> String => rust_i18n::locale() -> Arc<str>
```

`Arc<str>` is better than `Arc<String>`, what do you think?

**Comment by huacnlee** [comment]
Update this new benchmark result to `README.md`

```bash
t                       time:   [58.274 ns 60.222 ns 62.390 ns]
t_with_locale           time:   [55.395 ns 57.106 ns 59.081 ns]
t_with_args             time:   [167.46 ns 170.94 ns 175.64 ns]
t_with_args (str)       time:   [164.85 ns 165.91 ns 167.41 ns]
t_with_args (many)      time:   [444.04 ns 452.17 ns 463.44 ns]
t_with_threads          time:   [414.26 ns 422.97 ns 433.53 ns]
```

File: benches/bench.rs [hunk 1]
```diff
@@ -17,6 +17,26 @@ fn bench_t(c: &mut Criterion) {
```
Comment (huacnlee) [inline benches/bench.rs:17]: Update new value for this comment.

File: crates/macro/src/lib.rs [hunk 2]
```diff
@@ -17,14 +17,40 @@ impl Args {
         Ok(())
     }
 
+    fn consume_fallback(&mut self, input: syn::parse::ParseStream) -> syn::parse::Result<()> {
+        if let Ok(val) = input.parse::<LitStr>() {
+            self.fallback = Some(vec![val.value()]);
```
Comment (huacnlee) [inline crates/macro/src/lib.rs:22]:
```rs
return Ok(())
```

Return here first to avoid a `else`.

**Comment by varphone** [comment]
> ```
> rust_i18n::locale() -> String => rust_i18n::locale() -> Arc<str>
> ```
> 
> `Arc<str>` is better than `Arc<String>`, what do you think?

The AtomicPtr<str>

> ```
> rust_i18n::locale() -> String => rust_i18n::locale() -> Arc<str>
> ```
> 
> `Arc<str>` is better than `Arc<String>`, what do you think?

`AtomicStr` needs to hold the string content, which must be stored using a `String`.
Putting the `String` into `Arc` can achieve zero-copy reference, so returning `Arc<String>` directly is the most efficient way. 
If use `Arc<str>` as the return the value will produce a memory copy.

* https://doc.rust-lang.org/src/alloc/sync.rs.html#3315
* https://doc.rust-lang.org/src/alloc/sync.rs.html#3334

**Comment by huacnlee** [comment]
https://github.com/longbridgeapp/rust-i18n/releases/tag/v2.4.0

**Pull request merged**